File: UniEatsBackEnd/Models/Inventory.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace UniEatsBackEnd.Models
{
    public class Inventory
    {
        [Key]
        public int InventoryId { get; set; }

        [Required]
        public int ItemId { get; set; }

        [Range(0, int.MaxValue, ErrorMessage = "Stock added must be non-negative.")]
        public int StockAdded { get; set; }

        [Range(0, int.MaxValue, ErrorMessage = "Stock removed must be non-negative.")]
        public int StockRemoved { get; set; }

        [DatabaseGenerated(DatabaseGeneratedOption.Computed)]
        public int CurrentStock => StockAdded - StockRemoved;

        [Required]
        public DateTime UpdatedAt { get; set; } = DateTime.Now;
    }
}

File: UniEatsFrontEnd/UniEats/src/app/models/inventory.ts
import { FoodItem } from "./food-item";


export interface Inventory {
  inventoryId: number; // Unique identifier for the inventory record
  itemId: number; // ID of the associated food item
  stockAdded: number; // Stock added, must be non-negative
  stockRemoved: number; // Stock removed, must be non-negative
  currentStock?: number; // Computed property: stockAdded - stockRemoved
  updatedAt?: Date; // Defaults to the current date
  restockDate?: Date | null; // Optional restock date
  foodItem?: FoodItem; // Navigation property for the associated food item
}
